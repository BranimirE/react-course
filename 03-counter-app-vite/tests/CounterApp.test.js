const { render, screen, fireEvent } = require("@testing-library/react");
const { CounterApp } = require("../src/CounterApp");

describe('Tests in CounterApp', () => {
  const initialValue = 120;
  test('should match the snapshot', () => {
    const {container} = render(<CounterApp value={initialValue}/> );
    expect(container).toMatchSnapshot();
  });

  test('should show the initial value', () => {
    render(<CounterApp value={initialValue}/> );
    expect(screen.getByText(initialValue)).toBeTruthy();
  });

  test('should increase with +1 button', () => {
    render(<CounterApp value={initialValue}/> );
    fireEvent.click(screen.getByText('+1'));
    expect(screen.getByText(initialValue + 1)).toBeTruthy();
  });

  test('should decrease with -1 button', () => {
    render(<CounterApp value={initialValue}/> );
    fireEvent.click(screen.getByText('-1'));
    expect(screen.getByText(initialValue - 1)).toBeTruthy();
  });

  test('should reset the value with reset button click', () => {
    render(<CounterApp value={initialValue}/> );
    fireEvent.click(screen.getByText('+1'));
    fireEvent.click(screen.getByText('+1'));
    fireEvent.click(screen.getByText('+1'));
    fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}));
    expect(screen.getByText(initialValue)).toBeTruthy();
  })

})
