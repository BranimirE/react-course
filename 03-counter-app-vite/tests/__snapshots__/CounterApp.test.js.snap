// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in CounterApp should match the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    120
  </h2>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button>
    +1
  </button>
</div>
`;
